File: haskell/glue/src/Glue/Lib/Math/Power.hs
module Glue.Lib.Math.Power where

import Glue.Eval (Eval)
import Glue.Lib.Math.Power.Exp qualified as Exp
import Glue.Lib.Math.Power.Pow qualified as Pow
import Glue.Lib.Math.Power.Sqrt qualified as Sqrt
import Glue.Module (ModuleInfo, nativeModule)

power :: ModuleInfo Eval
power =
    nativeModule
        "ffi.math.power"
        [ ("exp", Exp.exp)
        , ("pow", Pow.pow)
        , ("sqrt", Sqrt.sqrt)
        ]

File: dart/glue/lib/src/lib/math/power.dart
import 'package:glue/src/lib/math/power/exp.dart';
import 'package:glue/src/lib/math/power/pow.dart';
import 'package:glue/src/lib/math/power/sqrt.dart';
import 'package:glue/src/module.dart';

/// Power module - exponential and power functions
/// Mirrors Haskell Glue.Lib.Math.Power exactly

/// The power module containing exponential and power operations
/// Mirrors Haskell Glue.Lib.Math.Power.power exactly
final ModuleInfo powerModule = nativeModule('ffi.math.power', [
  // Exponential function
  ('exp', exp),

  // Power function
  ('pow', pow),

  // Square root function
  ('sqrt', sqrt),
]);
